### ISSUE TITLE
Failed to load with two lora adapters

### ISSUE THREAD

**Issue by jarodxiangliu** [ggml-org/llama.cpp#18193]
### Name and Version

llama-server
version: 7476 (cdbada8d1)

### Operating systems

Windows

### GGML backends

CUDA

### Hardware

NVIDIA RTX A4000

### Models

qwen3-4b-thinking-2507, download from Huggingface

### Problem description & steps to reproduce

.\llama-server.exe -m ..\qwen3-4b-thinking-2507-q4_k_m.gguf --lora-scaled ..\LoraA.gguf:0.5 --lora-scaled ..\LoraB.gguf:0.5 --no-warmup -ngl 25

### Relevant log output

```shell
main: model loaded
main: server is listening on http://127.0.0.1:8080
main: starting the main loop...
srv  update_slots: all slots are idle
srv  log_server_r: request: GET /lora-adapters 127.0.0.1 200
slot get_availabl: id  3 | task -1 | selected slot by LRU, t_last = -1
slot launch_slot_: id  3 | task -1 | clearing cache for lora change. 0 loras -> 2 loras
slot launch_slot_: id  3 | task -1 | sampler chain: logits -> penalties -> dry -> top-n-sigma -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist
slot launch_slot_: id  3 | task 0 | processing task
slot update_slots: id  3 | task 0 | new prompt, n_ctx_slot = 27392, n_keep = 0, task.n_tokens = 2
slot update_slots: id  3 | task 0 | n_tokens = 0, memory_seq_rm [0, end)
slot update_slots: id  3 | task 0 | prompt processing progress, n_tokens = 2, batch.n_tokens = 2, progress = 1.000000
slot update_slots: id  3 | task 0 | prompt done, n_tokens = 2, batch.n_tokens = 2
D:/a/llama.cpp/llama.cpp/ggml/src/ggml.c:1700: GGML_ASSERT(obj_new) failed
ggml_new_object: not enough space in the context's memory pool (needed 1325760, available 1325392)
```

**Comment by CISC** [comment]
See other issue, needs the same fix.

**Comment by CISC** [comment]
It was actually already implemented there, you need to provide a comma-separated list to `--lora-scaled` now.

**Comment by jarodxiangliu** [comment]
@CISC Thanks for you reply first of all, I tried as your comment, with below command, it still failed
.\llama-server.exe -m ..\qwen3-4b-thinking-2507-q4_k_m.gguf --lora-scaled ..\vantage.gguf:0.5,..\privacyguard.gguf:0.5 -ngl 25 --no-warmup --verbose
But I could success with
.\llama-cli.exe -m ..\qwen3-4b-thinking-2507-q4_k_m.gguf --lora-scaled ..\vantage.gguf:0.5,..\privacyguard.gguf:0.5 -p "introduce yourself" -ngl 25 --no-warmup

below is my log:
que    start_loop: processing new tasks
que    start_loop: update slots
srv  update_slots: all slots are idle
que    start_loop: waiting for new tasks
srv          stop: all tasks already finished, no need to cancel
srv  log_server_r: request: GET /lora-adapters 127.0.0.1 200
srv  log_server_r: request:
srv  log_server_r: response: [{"id":0,"path":"..\\vantage.gguf","scale":0.5,"task_name":"","prompt_prefix":""},{"id":1,"path":"..\\privacyguard.gguf","scale":0.5,"task_name":"","prompt_prefix":""}]
srv  params_from_: Grammar:
srv  params_from_: Grammar lazy: false
res  add_waiting_: add task 0 to waiting list. current waiting = 0 (before add)
que          post: new task, id = 0/1, front = 0
que    start_loop: processing new tasks
que    start_loop: processing task, id = 0
slot get_availabl: id  3 | task -1 | selected slot by LRU, t_last = -1
slot        reset: id  3 | task -1 |
slot launch_slot_: id  3 | task -1 | clearing cache for lora change. 0 loras -> 2 loras
slot launch_slot_: id  3 | task -1 | launching slot : {"id":3,"n_ctx":57600,"speculative":false,"is_processing":false}
slot launch_slot_: id  3 | task -1 | sampler chain: logits -> penalties -> dry -> top-n-sigma -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist
slot launch_slot_: id  3 | task 0 | processing task
que    start_loop: update slots
srv  update_slots: posting NEXT_RESPONSE
que          post: new task, id = 1, front = 0
slot update_slots: id  3 | task 0 | new prompt, n_ctx_slot = 57600, n_keep = 0, task.n_tokens = 1
slot update_slots: id  3 | task 0 | n_tokens = 0, memory_seq_rm [0, end)
slot update_slots: id  3 | task 0 | prompt processing progress, n_tokens = 1, batch.n_tokens = 1, progress = 1.000000
slot update_slots: id  3 | task 0 | prompt done, n_tokens = 1, batch.n_tokens = 1
srv  update_slots: decoding batch, n_tokens = 1
D:/a/llama.cpp/llama.cpp/ggml/src/ggml.c:1700: GGML_ASSERT(obj_new) failed
clear_adapter_lora: call
set_adapter_lora: adapter = 000001FE51C58CB0, scale = 1.000000

**Comment by CISC** [comment]
> srv log_server_r: response: [{"id":0,"path":"..\vantage.gguf","scale":0.5,"task_name":"","prompt_prefix":""},{"id":1,"path":"..\privacyguard.gguf","scale":0.5,"task_name":"","prompt_prefix":""}]
> ...
> D:/a/llama.cpp/llama.cpp/ggml/src/ggml.c:1700: GGML_ASSERT(obj_new) failed clear_adapter_lora: call set_adapter_lora: adapter = 000001FE51C58CB0, scale = 1.000000

So it looks like it worked, but you're hitting some other problem, please create a new issue.

**Comment by jarodxiangliu** [comment]
@CISC Thanks for your clarify. I could create a new issue, but could you please help guide me that if I create new issue, which information should be provided, and possible issue? And as you can see, I also mentioned that, I could use llama-cli with the same two lora adapters, what make the difference?

**Comment by CISC** [comment]
> [@CISC](https://github.com/CISC) Thanks for your clarify. I could create a new issue, but could you please help guide me that if I create new issue, which information should be provided, and possible issue? And as you can see, I also mentioned that, I could use llama-cli with the same two lora adapters, what make the difference?

Hard to say, the error indicates an allocation error, but the reason why is not clear and requires debugging. Provide as much details as possible, including links to files used.
